cluster/engine.go

```diff
@@ -105,6 +107,7 @@ type Engine struct {
 	networks        map[string]*Network
 	volumes         map[string]*Volume
 	client          dockerclient.Client
+	apiClient       engineapi.APIClient
```

Comment:
@dongluochen can you elaborate on why `failureCount` and `updatedAt` should be part of the struct too?

Also, to clarify further, you're saying that these should be in a struct which is then part of the Engine struct right? I'm wondering if that should be part of this PR, WDYT?